### PR TITLE
#74 Form validation

### PR DESCRIPTION
**Pull request by adamgajzlerowicz**

### CODE REVIEW

File: frontend/src/views/RouteFormView.vue [hunk 1]
```diff
@@ -147,21 +144,32 @@ export default {
...
+        .reduce(makeErrorsPartial, {});
+
+      const { startMileage, endMileage } = this.route;
+
+      if (
+        !!startMileage
+        && !!endMileage
+        && parseInt(startMileage, 10) >= parseInt(endMileage, 10)
+      ) {
+        this.errors.startMileage = this.$t('common.start_mileage_error');
```
Comment (jacol12345) [inline frontend/src/views/RouteFormView.vue]: Errors are not translated if I switch language after validating form. We can live with that

Comment (jacol12345) [inline frontend/src/views/RouteFormView.vue]: There should be a separate error for a case when startMileage >= endMileage. It is confusing to see messages "Start mileage is incorrect" and "End mileage is incorrect" in such case.

File: frontend/src/views/services.js [hunk 2]
```diff
@@ -1,13 +1,23 @@
 const requiredFields = [
-  'date', 'car', 'description', 'from', 'destination', 'startMileage', 'endMileage',
+  'date',
+  'car',
+  'description',
```
Comment (jacol12345) [inline frontend/src/views/services.js]: A remotely related thing: text fields are validated as correct when filled with spaces. This one we need to fix.

File: frontend/src/views/RouteFormView.vue (hunk 1 above)
Comment (adamgajzlerowicz) [reply frontend/src/views/RouteFormView.vue]: I did consider that, but making error message that is way too specific might be just hard to read and confusing. Let's just make sure that the values are correct and user knows about incorrect ones.

**Pull request merged**